### PR TITLE
Unify transaction date & amount format parsing

### PR DESCRIPTION
**Pull request by weilu**
- both statement date and transaction date are now returned in the form of yyyy-mm-dd
- amount is returned as a float. Negative values indicate credit (e.g. income/refund)

### CODE REVIEW

File: cleaning/pdf-mining/pdftotxt.py [hunk 1]
```diff
@@ -18,8 +19,7 @@ def parse_statement_date(line, iterator):
     line_lower = line.lower()
     for clue in DATE_CLUES:
         if clue in line_lower:
-            statement_date = dateparser.parse(line_lower.split(clue)[-1],
-                                              languages=LANGUAGES)
```
Comment (ghost) [inline cleaning/pdf-mining/pdftotxt.py:22]: is this related to language detection?

File: cleaning/pdf-mining/pdftotxt.py [hunk 2]
```diff
@@ -29,11 +29,42 @@ def parse_statement_date(line, iterator):
...
+                statement_date = parse_date(groups[0])
                 if statement_date:
                     return statement_date
 
 
+# "(1,380.77)" and "1,380.77 CR" will translate to -1380.77
+def parse_amount(amount_str):
+    try:
+        amount_str = amount_str.lower().strip().replace(',', '')
+        cleaned = re.sub(r'\(|\)|cr', '', amount_str).strip()
```
Comment (ghost) [inline cleaning/pdf-mining/pdftotxt.py:41]: I was just checking if the value contains `(` and turns it into negative 😆

File: cleaning/pdf-mining/pdftotxt.py (hunk 1 above)
Comment (ghost) [reply cleaning/pdf-mining/pdftotxt.py:22]: where the `dateparser` come from? https://github.com/scrapinghub/dateparser? is this a global thing?

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: cleaning/pdf-mining/pdftotxt.py (hunk 2 above)
Comment (weilu) [reply cleaning/pdf-mining/pdftotxt.py:41]: (123.12) is what OCBC uses. DBS and UOB use a different format: 123.12 CR

File: cleaning/pdf-mining/pdftotxt.py (hunk 1 above)
Comment (weilu) [reply cleaning/pdf-mining/pdftotxt.py:22]: It's a python package. It's in [requirements.txt](https://github.com/weilu/cs4225/blob/money/requirements.txt#L14) and imported at the beginning of this file.
